internal/lock/lockfile.go

```diff
@@ -22,7 +23,7 @@ type File struct {
 	resolver
 
 	LockFileVersion string              `json:"lockfile_version"`
-	Packages        map[string]*Package `json:"packages"`
+	PackageMap      map[string]*Package `json:"packages"`
```

Comment:
You don't need to rename this. See this draft https://github.com/jetpack-io/devbox/pull/1113 

You can access conflicting functions on structs by specifying the embedded struct explicitly (e.g. lock.devboxProject.Packages())

That said, when embedded structs get this complicated it's probably better to get rid of them (and just make them explicit fields) (not something that needs to happen in this PR though).

Reply:
that draft was useful, thanks!

> That said, when embedded structs get this complicated it's probably better to get rid of them (and just make them explicit fields) (not something that needs to happen in this PR though).

yeah, that's what `savil/update-lockfile-v1` was about. But I thought it may receive pushback so took this more conservative approach. See #1120